src/datachain/data_storage/sqlite.py

```
@@ -150,14 +152,21 @@ def _connect(
             db.execute("PRAGMA journal_mode = WAL")
             db.execute("PRAGMA synchronous = NORMAL")
             db.execute("PRAGMA case_sensitive_like = ON")
+
+            max_variable_number = 999  # minimum in old SQLite versions
```

where does it come from? I mean 999 ... (see too small)

---

That's the lowest ever value that was present for that option. It was in SQLite versions lower than `3.32.0`.

---

If we always catch it - let's remove and raise (it is better vs silently using such a small value)

---

Isn't it better to use that smaller number then to raise Exception? ... dataset will still be pulled, just slower.

---

Yeah, but it might become always slow silently in case we make mistake below ... better to be aware about this

---

If pragma exists everywhere / always - what is the issue with it then?